### DISCUSSION TITLE
How long it took to run the nasdaq example?

### DISCUSSION THREAD

**Discussion by panyusheng1222** [Boulder-Investment-Technologies/lppls#57] [Q&A]
It burned my laptop

**Comment by Joshwani** [comment]
Yeah, it takes a while. You can try changing the optimizer from 'Nelder-Mead' to 'SLSQP', @nightvision04 pointed out that it's quite a bit faster that way. Otherwise, run in cloud with many cores, can be pricey though.

**Reply by nightvision04** [reply]
@Joshwani I must apologize. While method='SLSQP' is faster and worked in the unit tests, it actually isn't very stable. I've found the best tradeoff between speed and stability to be:

`minimize(method='Nelder-Mead',
            options={'xatol': 1e-3})`
            
In fact, after extensive testing against different CMA-ES optimizers, the above settings were still far superior in both speed and stability.

**Reply by nightvision04** [reply]
Though SLSQP was fast, estimates ended up snapping to either side of the bounded range (producing a lot of false positives). This is because of the assumption that error is quadratic. But in reality, error is non-linear, but _also_ is not necessarily quadratic.

**Reply by nightvision04** [reply]
This issue went away when I went back to Nelder-Mead. I tried integrating the 3 popular CMA-ES libraries and found them to be fairly reliable but slow. Ultimately, my tests showed that Nelder-Mead performed at near-deterministic reliability (with the above hyperparameter) but much faster than CMA-ES and with fewer false positives.
